test/functional/connection.test.js

```diff
@@ -273,4 +274,31 @@ describe('Connection', function () {
       done();
     }
   });
+
+  it('should be able to connect again after close', function () {
+    return withClient.bind(this)((client, done) => {
```

Comment:
`withClient` is totally wonk right now, but because you're not passing in any extra options to it, it can be used like this:

```
it('should be able to connect again after close', withClient(function (client, done) {
```

Note you can't use arrow function to maintain mocha's `this` scoping.